internal/factsengine/gatherers/corosynccmapctl.go

```diff
@@ -45,6 +44,38 @@ func NewCorosyncCmapctlGatherer(executor utils.CommandExecutor) *CorosyncCmapctl
 	}
 }
 
+func corosyncCmapctlOutputToMap(corosyncCmapctlOutput string) *entities.FactValueMap {
+	outputMap := &entities.FactValueMap{Value: make(map[string]entities.FactValue)}
+	var cursor *entities.FactValueMap
+
+	for _, line := range strings.Split(corosyncCmapctlOutput, "\n") {
+		cursor = outputMap
+
+		value := strings.Split(line, "= ")[1]
+
+		pathAsString := strings.Split(line, " (")[0]
+		path := strings.Split(pathAsString, ".")
+
+		for i, key := range path {
+			currentMap := cursor
+
+			if i == len(path)-1 {
+				currentMap.Value[key] = entities.ParseStringToFactValue(value)
+
+				break
+			}
+
+			if _, found := currentMap.Value[key]; !found {
+				currentMap.Value[key] = &entities.FactValueMap{Value: make(map[string]entities.FactValue)}
+			}
+
+			cursor = currentMap.Value[key].(*entities.FactValueMap) //nolint:forcetypeassert
```

Comment:
We shouldn't hide this assertion linting error. We should check if it's ok, and return an error if it cannot be asserted.

Reply:
The type assertion here could be ignored since we are 100% sure that the value is a `FactValueMap`
We could also do the error handling, as you proposed, but keep in mind that it's impossible to happen.

@CDimonaco what is the idiomatic way to do this?

Reply:
I don't know...
We have done the assertion checking always in this cases, even knowing that it would get the same type 100% cases.
Anyway, I won't discuss on this